src/app/flightsearch/flightsearch.component.ts

```diff
@@ -0,0 +1,538 @@
+        this.flightsData = [];
+        this.viewData = [];
+        this.noResultsErrorMsg = "We could not find any flights with your search results. Please try again with different parameters";
+        return;
+      }
+      else{
+
+        this.flightsData = processedData;
+        //Tie it into the viewData list
+        this.viewData = [...this.flightsData];
+        this.sortData();
+      }
+    }
+
+    //So round trip flights
+    if (this.tripRB != "One-Way"){
+      let destinationToOrigin = res["Destination to origin"];
+
+      if(destinationToOrigin.length == 0){
+        this.returnFlightsData = [];
+        this.returnTripErrorMsg = "We could not find any flights that return from your destination. Do you still want to view your results?";
+        return;
+      }
+      else
+      { 
+        let processedData = this.standardizeResults(destinationToOrigin);
+        if (processedData.length == 0){
+          this.returnFlightsData = [];
+          this.returnTripErrorMsg = "We could not find any flights that return from your destination. Do you still want to view your results?";
+          return;
+        }
+        else{
+          //Do not make it viewdata; just need to store it
+          this.returnFlightsData = processedData;
+        }
+      }
+    }
+  }
+
+  standardizeResults(data: Array<any>): Array<any>{
+
+    //Create number formatter
+    let formatter = new Intl.NumberFormat("en-US", {
+      style: "currency",
+      currency: "USD",
+    });
+
+    let getCorrectFlights:any = [];
+    if (this.nonStopRB == "Multihop"){
+      data.forEach((element: any) => {
+        let tempObject:any = {};
+
+        //Only one flight
+        if (element.length == 1){
+           tempObject = this.nonStopFlight(element);
+        }
+
+        //More than one flight
+        else{
+          let tempPrice = 0;
+          let tempLoyaltyPoints = 0;
+          let tempDuration = 0;
+          let tempIataId = "";
+          let tempCities = [];
+
+          //Go through each element
+          element.forEach((flight: any, index:number) => {
+            tempLoyaltyPoints += flight.possibleLoyaltyPoints;
+            tempPrice += flight.seats[0].price;
+            if (index == 0){
+              tempIataId += flight.origin.iataId + " to " + flight.destination.iataId;
+            }
+            else{
+              tempIataId += " to " + flight.destination.iataId;
+            }
+            tempCities.push(flight.origin.servicingArea.servicingArea);
+            tempDuration += new Date(flight.approximateDateTimeEnd).getTime() - new Date(flight.approximateDateTimeStart).getTime();
+          });
+
+          let economySeats = element[0].seats.filter((x:any) => x.seatClass == "ECONOMY" && x.seatStatus=="AVAILABLE");
+          let businessSeats = element[0].seats.filter((x:any) => x.seatClass == "BUSINESS" && x.seatStatus=="AVAILABLE")
+          tempObject["economy"] = economySeats.length;
```

Comment:
There's some major duplication starting here and again at line 338. There's also a ton going on so it's difficult to parse and understand. Definitely consider extracting this into a helper function that returns the aforementioned `FlightData` object.